adds chapter on emax covariate modelling

djnavarro opened the pull request:
Main changes in this PR:

- There is a new chapter 10 illustrating emax model development with brms
- The data generation chapter (now chapter 11) has two new data sets 

The emax modeling chapter covers:

- Core syntax: `brmsformula()`, `brmsfamily()`, `prior()`, and `brm()`
- Examples with continuous-response emax data and binary-response emax data
- Building models with and without covariates
- Examples of visualizing model predictions
- Model comparison with LOO-CV

Closes #1

djnavarro commented:
Hi @yoshidk6, 

This is a first draft for a brms/emax modelling chapter. It's still a little rough around the edges, but the core concepts are there. I think it is at the point where review would be useful, and have some thoughts/questions about what might be needed at this stage:

- The current draft only considers regular emax models with gamma=1. Elsewhere in the book there are examples that use the full sigmoidal emax model, and it may be worth illustrating that in this context also
- Covariate modeling examples only show the case when covariates are placed on the baseline response, and it may be worth showing examples where covariates are placed on other parameters in the emax structural model
- The chapter doesn't go into much detail on how to use tidybayes/posterior tools with brms models. Some of that is implied in the data visualisation, but not much. I'm not sure if this is really needed though
- The data visualisations don't show credible intervals or prediction intervals, only the mean predictions. This seems like a big omission, but I'm not sure how much detail we want to include here?
- Model diagnostics are almost entirely omitted, except insofar as the default output prints some of this information; and the model comparison section is very minimal. More detail may be needed here?

Best, Danielle

yoshidk6 commented:
Hi Danielle, I really appreciate your efforts, this is already looking great!!
Please see my comments below. They are ordered in a roughly the order of priority in my gut feel

- I think it makes sense to split the current document into two, one showcasing brms based workflow (similar to https://genentech.github.io/BayesERbook/notebook/binary/workflow_wo_package.html) and the other one showcasing covariate modeling. 
- > Covariate modeling examples only show the case when covariates are placed on the baseline response, and it may be worth showing examples where covariates are placed on other parameters in the emax structural model

    - Fully agree, effect on Emax or EC50 would be nice
 
- For priors on binary Emax model, suggest to use normal(0, 1.5) for E0 and Emax (based on Statistical Rethinking https://discourse.mc-stan.org/t/default-priors-for-logistic-regression-coefficients-in-brms/13742)
- In the first page after split, we can probably expand on how to set priors? (not top priority)

- > The chapter doesn't go into much detail on how to use tidybayes/posterior tools with brms models. Some of that is implied in the data visualisation, but not much. I'm not sure if this is really needed though
- > The data visualisations don't show credible intervals or prediction intervals, only the mean predictions. This seems like a big omission, but I'm not sure how much detail we want to include here?

```r
continuous_base_fit |> 
  epred_draws(newdata = tibble(exposure = seq(0, 50000, 1000))) |> 
  median_qi() %>% 
  ggplot(mapping = aes(exposure, .epred)) + 
  geom_path() + 
  geom_ribbon(aes(ymin = .lower, ymax = .upper), alpha = 0.5) +
  geom_point(data = d_example_emax_3cov, aes(y = response))
```

- > Model diagnostics are almost entirely omitted, except insofar as the default output prints some of this information; and the model comparison section is very minimal. More detail may be needed here?
  - I think what we have is fairly good at least for now

yoshidk6 commented:
Down the road I also want to incorporate data generation into `BayesERtools` so that the readers don't have to download the csv file.. Submitted separate issues (https://github.com/Genentech/BayesERbook/issues/6 https://github.com/Genentech/BayesERtools/issues/6)

djnavarro commented:
Hi @yoshidk6, sorry it took me a while to make the changes. The revised PR:

- splits the brms modelling into two chapters: chapter 10 covers simple Emax models with brms, chapter 11 discusses covariate modelling and model selection with LOO-CV
- parameter priors updated as suggested
- minor updates to the simulated data to make model selection results clearer 
- `median_qi()` is now used to display intervals around the regression line
- new example in section 10.2 fits the sigmoidal Emax model with `gamma` as an estimated quantity
- new example in section 11.3 places covariates on `emax` as well as `e0`
- LOO-CV example slightly updated to include base model, model with one covariate, and a model with three covariates; showing that the one-covariate model works best  

If this looks good to merge I'm happy to turn to #6 next? But of course let me know if there's a higher priority :)

yoshidk6 commented:
Thank you so much for this! 
As I'm out of office now, I've only taken a quick look, but it looks really good.
Let me take a deeper look sometime next week.

In the meantime,

- Yes it would be great if you can start on #6
- Can you add brms caching setting as you can see here? https://github.com/RCONIS/tgi-os-training/blob/main/session-tgi/_setup_and_load.qmd#L23-L30

djnavarro commented:
Done! Added brms caching on the two relevant chapters. I'll start work on adding data generation functions to BayesERtools :)

yoshidk6 commented:
Thank you so much! Had chance to look through the code and made some suggested changes. 
I couldn't push them to this repo as I don't have a write privilege, could you check the files and push the changes from your end, if these changes make sense?
[qmd_edits.zip](https://github.com/user-attachments/files/19702638/qmd_edits.zip)

Also it would be great if you can add note on how the priors are defined for covariate effects. 

After these I'll close the issue and merge this pull request. Thanks!

djnavarro commented:
Agreed that those changes make sense. I've incorporated your edits into the PR, and added a brief comment highlighting the fact that priors are applied to all covariate coefficients when a covariate models is specified. Obviously we'll need revisit once https://github.com/Genentech/BayesERtools/pull/7 is merged, but I agree it makes sense to merge the current version and open a new PR once the package data are updated

yoshidk6 commented:
Perfect, thank you!!
